browser/js/common/directives/graphs/barChart.js

```diff
@@ -22,10 +22,22 @@ app.directive('barChart', function(d3Service, $window, DataFactory) {
           scope.$watch(function() {
             return angular.element($window)[0].innerWidth;
           }, function() {
-            scope.render(scope.data);
+            scope.render();
```

Comment:
Geoff/Corey,

This all looks good.  I see what you mean about the refactoring. 
One thing, I was looking at Omri's comments from yesterday on the line-graph, and he mentioned that there may be an issue with re-rendering if we assign scope.columns or scope.data to another variable in the directive.   
Here was his comment

'Why is scope.cols coming in as a comma separated string, why not just pass it in as an array? Also keep in mind that if scope.dataColumns is simply the result of having split scope.cols right here right now, then scope.dataColumns is not two-way-bound to whatever scope.cols references. So if scope.cols changes (due to some ng-click or whatnot) scope.dataColumns will not update—unless this template re-renders, at which point this linking function will run again and this scope.dataColumns will get re-assigned.'

I'm not sure if your bar and scatter plot directives did not reassign the scope variables (or if you already looked at this), but maybe this would take care of the re-rendering, rather than having to do a $scope.$watch?